cozy/ui/widgets/seek_bar.py

```diff
@@ -19,15 +19,25 @@ def __init__(self, **kwargs):
 
         self.progress_scale.connect("value-changed", self._on_progress_scale_changed)
 
-        self._progress_scale_gesture = Gtk.GestureClick()
-        self._progress_scale_gesture.connect("pressed", self._on_progress_scale_press)
-        self._progress_scale_gesture.connect("end", self._on_progress_scale_release)
-        self.progress_scale.add_controller(self._progress_scale_gesture)
-
-        self._progress_scale_key = Gtk.EventControllerKey()
-        self._progress_scale_key.connect("key-pressed", self._on_progress_scale_press)
-        self._progress_scale_key.connect("key-released", self._on_progress_scale_release)
-        self.progress_scale.add_controller(self._progress_scale_key)
+        # HACK: Using a GtkGestureClick here is not possible, as GtkRange's internal
+        # gesture controller claims the button press event, and thus the released signal doesn't get emitted.
+        # Therefore we get its internal GtkGestureClick, and add our handlers to that.
+        # Hacky workaround from: https://gitlab.gnome.org/GNOME/gtk/-/issues/4939
+        # Ideally GtkRange would forward these signals, so we wouldn't need this hack
+        # TODO: Add these signals to Gtk and make a MR?
+        for controller in self.progress_scale.observe_controllers():
+            if isinstance(controller, Gtk.GestureClick):
+                click_gesture = controller
```

Comment:
Is it possible there would be no matching controller? If so, then `click_gesture` would be unbound below?

Reply:
Only if the `GtkGestureClick` was removed from the `GtkRange` C code, which is very unlikely, so I didn't bother.